.github/workflows/publish.yml

```diff
@@ -50,7 +50,7 @@ jobs:
       - name: Set up JDK 11
         uses: olafurpg/setup-scala@v10
         with:
-          java-version: adopt@1.11.0-9
+          java-version: adopt@1.16.0-1
```

Comment:
Hmm, do we want to though, until we explicitly support a newer JDK?

If we want to we should probably change the step name above as well ;)

Reply:
Yeah, it's questionable whether this makes sense yet. I'm OK with shelving this until these become more common. My main motivation was to see if #30356 would be fixed with a newer Java version (it isn't, probably a genjavadoc problem TBQH).

Reply:
One motivation could perhaps be if the generated javadoc has a lot nicer UI (haven't checked)

Reply:
they don't